test/datatable.jl

```diff
@@ -234,140 +232,142 @@ module TestDataTable
 
     @test append!(DataTable(A = 1:2, B = 1:2), DataTable(A = 3:4, B = 3:4)) == DataTable(A=1:4, B = 1:4)
-    dt = DataTable(A = NullableArray(1:3), B = NullableArray(4:6))
-    @test all(c -> isa(c, NullableArray), categorical!(deepcopy(dt)).columns)
-    @test all(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [1,2]).columns)
-    @test all(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [:A,:B]).columns)
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [:A]).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), :A).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [1]).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), 1).columns) == [1]
+    dt = DataTable(A = 1:3, B = 4:6)
+    @test all(c -> isa(c, Vector), categorical!(deepcopy(dt)).columns)
+    @test all(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [1,2]).columns)
+    @test all(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [:A,:B]).columns)
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [:A]).columns) == [1]
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), :A).columns) == [1]
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [1]).columns) == [1]
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), 1).columns) == [1]
 
     @testset "unstack nullable promotion" begin
         dt = DataTable(Any[repeat(1:2, inner=4), repeat('a':'d', outer=2), collect(1:8)],
                        [:id, :variable, :value])
         udt = unstack(dt)
-        @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
-        @test udt == DataTable(Any[Nullable[1, 2], Nullable[1, 5], Nullable[2, 6],
-                                   Nullable[3, 7], Nullable[4, 8]], [:id, :a, :b, :c, :d])
-        @test all(typeof.(udt.columns) .== NullableVector{Int})
+        #FIXME
+        # @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
+        @test udt == DataTable(Any[[1, 2], [1, 5], [2, 6],
+                                   [3, 7], [4, 8]], [:id, :a, :b, :c, :d])
+        @test all(typeof.(udt.columns) .== Vector{?Int})
         dt = DataTable(Any[categorical(repeat(1:2, inner=4)),
                            categorical(repeat('a':'d', outer=2)), categorical(1:8)],
                        [:id, :variable, :value])
         udt = unstack(dt)
-        @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
-        @test udt == DataTable(Any[Nullable[1, 2], Nullable[1, 5], Nullable[2, 6],
-                                   Nullable[3, 7], Nullable[4, 8]], [:id, :a, :b, :c, :d])
+        # @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
```

Comment:
@nalimilan, this test fails w/ error
```julia
julia> unstack(dt, :variable, :value)
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type CategoricalArrays.CategoricalValue{Int64,UInt32}
This may have arisen from a call to the constructor CategoricalArrays.CategoricalValue{Int64,UInt32}(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] copy!(::IndexLinear, ::Array{Union{CategoricalArrays.CategoricalValue{Int64,UInt32}, Nulls.Null},1}, ::IndexLinear, ::CategoricalArrays.CategoricalArray{Int64,1,UInt32}) at ./abstractarray.jl:648
 [2] Array{Union{CategoricalArrays.CategoricalValue{Int64,UInt32}, Nulls.Null},1}(::CategoricalArrays.CategoricalArray{Int64,1,UInt32}) at ./sysimg.jl:24
 [3] nullable!(::DataTables.DataTable, ::Symbol) at /Users/jacobquinn/.julia/v0.6/DataTables/src/datatable/datatable.jl:732
 [4] nullable!(::DataTables.DataTable, ::Array{Symbol,1}) at /Users/jacobquinn/.julia/v0.6/DataTables/src/datatable/datatable.jl:737
 [5] unstack(::DataTables.DataTable, ::Int64, ::Int64) at /Users/jacobquinn/.julia/v0.6/DataTables/src/abstractdatatable/reshape.jl:235
 [6] unstack(::DataTables.DataTable, ::Symbol, ::Symbol) at /Users/jacobquinn/.julia/v0.6/DataTables/src/abstractdatatable/reshape.jl:222
```
I don't think we want to bother adding a `nullable!` method for `CategoricalArray` => `NullableCategoricalArray` if you're going to be updating the null vector to a union anyway. Let's just keep a bread crumb here to come back to after you make the change and make sure this works.